wdl-engine/src/http.rs

```diff
@@ -338,6 +350,13 @@ impl Downloader for HttpDownloader {
                 continue;
             }
 
+            let permit = self
```

Comment:
```rust
let permit = self
    .semaphore
    .acquire()
    .await
    .expect("semaphore should not be closed");
```

This should be sufficient here, you don't need anything more than a standard semaphore permit.